mypy/freetree.py

```diff
@@ -0,0 +1,14 @@
+"""Generic node traverser visitor"""
+
+from mypy.traverser import TraverserVisitor
+from mypy.nodes import Block, MypyFile
+
+
+class TreeFreer(TraverserVisitor):
+    def visit_block(self, block: Block) -> None:
+        super().visit_block(block)
```

Comment:
Would clearing just the top-level blocks be sufficient here, instead of recursively clearing?

Reply:
We need to go recursively or we'll miss nested structures that appear in the symbol table.